components/mod-event/EventItem.tsx

```diff
@@ -31,6 +31,38 @@ const LinkToAuthor = ({
   )
 }
 
+// Utility function to detect and replace links with <a> tags
+const wrapLinksInText = (text: string): JSX.Element[] => {
+  // Regular expression to match URLs
+  const urlRegex = /(https?:\/\/[^\s]+)/g
+
+  // Split text into parts, with URLs as matches
+  const parts = text.split(urlRegex)
+
+  return parts.map((part, index) => {
+    if (urlRegex.test(part)) {
```

Comment:
Global (`/g`) regexes are stateful when used with `.test()`, taking into account the last matched index.  I think that this could lead to missed matches here.

Reply:
For example:
```js
const regex = /a/g
regex.test('a') // true
regex.test('a') // false
```

Reply:
again, not sure how accurate I am with this but it seems to be only an issue when applied to same string and since we're applying it on different `parts` I think this works fine...? I did test this with various possible inputs where it may fail but it seems to work fine but happy to make changes if we find issues down the road.
<img width="425" alt="Screenshot 2024-12-23 at 22 43 27" src="https://github.com/user-attachments/assets/db091b44-bcd2-4689-94c9-3c147af6bcf1" />